src/imap/mod.rs

```diff
@@ -1140,54 +1140,54 @@ impl Imap {
                 }
             };
 
+            let mut delimiter = ".".to_string();
+            let mut delimiter_is_default = true;
             let mut sentbox_folder = None;
             let mut mvbox_folder = None;
+            let mut fallback_folder = get_fallback_folder(&delimiter);
 
-            let mut delimiter = ".".to_string();
-            if let Some(folder) = folders.next().await {
+            while let Some(folder) = folders.next().await {
                 let folder = folder.map_err(|err| Error::Other(err.to_string()))?;
+                info!(context, "Scanning folder: {:?}", folder);
+
+                // Update the delimiter iff there is a different one, but only once.
```

Comment:
typo: iff

Reply:
iff is a shorthand for `if and only if`, not a typo :)

Reply:
didn't know that, thanks for explanations!